lib/sycamore/sycamore/document.py

```diff
@@ -0,0 +1,388 @@
+from abc import abstractmethod, ABC
+from enum import Enum
+
+
+class Category(Enum):
+    Caption = 1
+    Footnote = 2
+    Formula = 3
+    ListItem = 4
+    PageFooter = 5
+    PageHeader = 6
+    Picture = 7
+    SectionHeader = 8
+    Table = 9
+    Text = 10
+    Title = 11
+    Section = 101
+    Group = 102
+
+
+class Node:
+    def __init__(self, node_id, metadata):
+        self._node_id = node_id
+        self._metadata = metadata
+
+    def node_id(self):
+        return self._node_id
+
+    @abstractmethod
+    def category(self):
+        pass
+
+    def metadata(self):
```

Comment:
Is this intended to replace properties? What is the signature, `dict[str, Any]`? Also, how would you update it if you want to add properties from like entity extraction or something?

Reply:
yes, it's dict[str, Any]. Maybe not get your point, a visitor could help here.